Library/Homebrew/cask/artifact/shim_script.rb

```diff
@@ -0,0 +1,42 @@
+# typed: true # rubocop:todo Sorbet/StrictSigil
+# frozen_string_literal: true
+
+require "cask/artifact/binary"
+
+module Cask
+  module Artifact
+    # Artifact corresponding to the `shim_script` stanza.
+    class ShimScript < Binary
+      DEFAULT_ARGS = ["$@"].freeze
+
+      sig {
+        params(
+          cask:    Cask,
+          source:  T.any(String, Pathname),
+          target:  T.nilable(T.any(String, Pathname)),
```

Comment:
May just have same constraint as `wrapper` to restrict usage to what we need in `homebrew/cask`, i.e. all targets are in `HOMEBREW_PREFIX/bin`
```suggestion
          target:  T.nilable(String),
```